Logging in a newly created coach user returns a blank screen

<!--
Instructions:
 * Fill out the sections below, replace …'s with information about your issue
 * Use the 'preview' function above this text box to verify formatting before submitting
-->

## Observed behavior
<!--
Description of the behavior that was observed, including screenshots or other references when applicable
-->
Creating a coach user and logging in using the created credentials returns a blank screen

https://github.com/learningequality/kolibri/assets/5203639/d7b82cea-8bc2-404a-a310-56855c5cb420

## Errors and logs
<!--
Relevant logs from:
 * the command line
 * ~/.kolibri/logs/kolibri.txt
 * the browser console

Please wrap errors in triple backticks for clean formatting like this:
```
01:10 info: something happened
01:12 error: something bad happened
```
-->
```
vue-router.esm.js:2316 Error: Missing required 'facilityId' argument
    at Store.setClassList (pluginModule.js:70:1)
    at Array.wrappedActionHandler (vuex.esm.js:851:1)
    at Store.dispatch (vuex.esm.js:516:1)
    at Store.boundDispatch [as dispatch] (vuex.esm.js:406:1)
    at Object.handler [as CoachClassListPage] (index.js:30:1)
    at Router._hook (router.js:22:1)
    at iterator (vue-router.esm.js:2362:1)
    at step (vue-router.esm.js:2087:1)
    at vue-router.esm.js:2088:1
    at vue-router.esm.js:2384:1
```

## Expected behavior
<!--
Description of what behavior was expected but did not occur
-->
Coach user should not be returned with a blank screen on login

## User-facing consequences
<!--
Implications and real-world consequences for learners, coaches, admins, and other users of the application
-->
A coach user is not able to use kolibri

## Steps to reproduce
<!--
Precise steps that someone else can follow in order to see this behavior
-->

1. Login into a full device as superuser/admin
2. Create a coach user
3. Logout of superuser/admin
4. Login with created coach user

## Context
<!--
Tell us about your environment, including:
 * Kolibri version
 * Operating system
 * Browser
-->
`Kolibri 0.16.0a14.dev0+git.124.g726134e8 ` 
MacOs
Chrome Version 114.0.5735.90 (Official Build) (arm64)

--- a/kolibri/plugins/coach/assets/src/modules/pluginModule.js
+++ b/kolibri/plugins/coach/assets/src/modules/pluginModule.js
@@ -63,15 +63,13 @@ export default {
   },
   actions: {
     setClassList(store, facilityId) {
-      const activeFacilityId =
-        store.state.core.facilities.length === 1 ? store.getters.userFacilityId : facilityId;
-      if (!activeFacilityId) {
+      if (facilityId) {
         throw new Error("Missing required 'facilityId' argument");
       }
       store.commit('SET_DATA_LOADING', true);
       store.commit('SET_CLASS_LIST', []); // Reset the list if we're loading a new one
       return ClassroomResource.fetchCollection({
-        getParams: { parent: activeFacilityId, role: 'coach' },
+        getParams: { parent: facilityId, role: 'coach' },
       })
         .then(classrooms => {
           store.commit('SET_CLASS_LIST', classrooms);

--- a/kolibri/plugins/coach/assets/src/modules/pluginModule.js
+++ b/kolibri/plugins/coach/assets/src/modules/pluginModule.js
@@ -63,7 +63,7 @@ export default {
   },
   actions: {
     setClassList(store, facilityId) {
-      if (facilityId) {
+      if (!facilityId) {
         throw new Error("Missing required 'facilityId' argument");
       }
       store.commit('SET_DATA_LOADING', true);

--- a/kolibri/plugins/coach/assets/src/routes/index.js
+++ b/kolibri/plugins/coach/assets/src/routes/index.js
@@ -27,7 +27,10 @@ export default [
     component: CoachClassListPage,
     handler(toRoute) {
       store.dispatch('loading');
-      store.dispatch('setClassList', toRoute.query.facility_id).then(
+      // if user only has access to one facility, facility_id will not be accessible from URL,
+      // but always defaulting to userFacilityId would cause problems for multi-facility admins
+      const facilityId = toRoute.query.facility_id || store.getters.userFacilityId;
+      store.dispatch('setClassList', facilityId).then(
         () => {
           if (!store.getters.classListPageEnabled) {
             // If no class list page, redirect to

